light/rpc/client.go

```diff
@@ -579,8 +579,16 @@ func (c *Client) Tx(ctx context.Context, hash []byte, prove bool) (*ctypes.Resul
 		return nil, err
 	}
 
-	// Validate the proof.
-	return res, res.Proof.Validate(l.DataHash)
+	// Check if the proof is correctly constructed.
+	if err := res.Proof.Validate(); err != nil {
+		return nil, err
+	}
+
+	// Verify the proof
+	if !res.Proof.VerifyProof(l.DataHash) {
+		return nil, fmt.Errorf("invalid transaction shares proof")
```

Comment:
[question] is res.Proof guaranteed to be a shares proof and not a row proof?

Reply:
the transaction proof is a share proof (a share proof includes internally a row proof), why would it contain a row proof instead?

Reply:
idk, just asking b/c I didn't know 

> the transaction proof is a share proof

Reply:
it's a share proof in the consensus network because we prove it against a data root. In the Celestia-node, it is only a share to row root NMT proof, because it's proven to the data availability header.